src/v2/manifest/mod.rs

```diff
@@ -286,28 +286,45 @@ fn evaluate_media_type(
     }
 }
 
-fn build_accept_headers() -> header::HeaderMap {
-    header::HeaderMap::from_iter(
-        [
-            // accept header types and their q value, as documented in
-            // https://tools.ietf.org/html/rfc7231#section-5.3.2
-            (mediatypes::MediaTypes::ManifestV2S2, 0.5),
-            (mediatypes::MediaTypes::ManifestV2S1Signed, 0.4),
-            // TODO(steveeJ): uncomment this when all the Manifest methods work for it
-            // mediatypes::MediaTypes::ManifestList,
-        ]
-        .iter()
-        .filter_map(|(ty, q)| {
-            match header::HeaderValue::from_str(&format!("{}; q={}", ty.to_string(), q)) {
-                Ok(header_value) => Some((header::ACCEPT, header_value)),
-                Err(e) => {
-                    let msg = format!("failed to parse HeaderValue from str: {}:", e);
-                    error!("{}", msg);
-                    None
+fn build_accept_headers(registry: &str) -> header::HeaderMap {
+    use strum::AsStaticRef;
+
+    // GCR incorrectly parses `q` parameters, so we use special Accept for it.
+    // Bug: https://issuetracker.google.com/issues/159827510.
+    // TODO: when bug is fixed, this workaround should be removed.
+    let no_q = registry.ends_with("://gcr.io") || registry.ends_with(".gcr.io");
```

Comment:
The protocol [cannot be specified in a reference](https://github.com/docker/distribution/blob/0d3efadf0154c2b8a4e7b6621fff9809655cc580/reference/reference.go#L4), so we should never encounter `://` here. Did you actually encounter this somehow?

If I'm not mistaken, this should work:

```suggestion
    let no_q = registry == "gcr.io" || registry.ends_with(".gcr.io");
```

Reply:
Well, I'm quite bad in naming :)
This functions in fact receives registry API endpoint, which [contains](https://github.com/camallo/dkregistry-rs/blob/d6dc42dc8e729fe6c5f8f4792e8b445f4c29eae4/src/v2/config.rs#L74) schema too.
I can either rename this parameter, or somehow change `Client` (for example it can store registry hostname and scheme separately).

Reply:
I understand the confusion now and I added another suggestion to complement the existing one. I'd prefer to not deal with the schema here.